### ISSUE TITLE
Support external SSLContext

### ISSUE THREAD

**Issue by fdulger** [CloudBees-community/syslog-java-client#30]
In our particular case, we need to use SSL but bypass the certificate check. When TcpSyslogMessageSender is set to use SSL it creates a secure socket using default SSLSocketFactory. It could be better if I can provide a custom SSLContext and that is used to open the socket.

In TcpSyslogMessageSender:
```
ensureSyslogServerConnection(){
...
    if(sslContext == null) {
        socket = SSLSocketFactory.getDefault().createSocket();
    } else {
        socket = sslContext.getSocketFactory().createSocket();
    }
...
}

...

public synchronized void setSSLContext(SSLContext sslContext) {
    this.sslContext = sslContext; 
}    
public synchronized SSLContext getSSLContext() {
    return this.sslContext; 
}
```

**Comment by KevUKHD** [comment]
I'll second that.
This would be very useful to have.
For my usecase, I need to add my own Keystore / Certificate to the sslContext.

**Comment by fdulger** [comment]
@KevUKHD This is available since 1.1.7
